merge_duplicate_contacts/wizard/merge_contact.py

```diff
@@ -153,66 +153,42 @@ def _action_new_next_screen(self):
     def _generate_query(self, fields, maximum_group=100):
         final_querry = "SELECT min(id), array_agg(id) FROM res_partner "
 
-        where_querries = []
-        group_by_querries = []
-
-        commercial_email_domains = (
-            " ('aikq.de','aol.com','aol.de','arcor.de',"
-            "'bluewin.ch','compuserve.com','dismail.de','disroot.org','duck.com','email.de',"
-            "'ewe.net','ewetel.net','fastmail.com','fastmail.de','fastmail.fm','fastmail.net',"
-            "'free.fr','freenet.de','gmail.com','gmx.at','gmx.ch','gmx.com','gmx.de','gmx.eu',"
-            "'gmx.fr','gmx.info','gmx.li','gmx.org','gmxpro.de','gmx-topmail.de',"
-            "'googlemail.com','hotmail.ch','hotmail.co.uk','hotmail.com','hotmail.de',"
-            "'hotmail.es','hotmail.fr','hotmail.it','hush.com','hushmail.com','icloud.com',"
-            "'jpberlin.de','kabelmail.de','laposte.net','lavabit.com','librem.one',"
-            "'live.co.uk','live.com','live.de','live.fr','live.nl','mac.com','mail.de',"
-            "'mailbox.org','mailfence.com','me.com','meineinkauf.ch','msn.com',"
-            "'mykolab.ch','mykolab.com','netcologne.de','online.de','onlinehome.de',"
-            "'orange.fr','outlook.com','outlook.de','outlook.es','outlook.fr',"
-            "'posteo.at','posteo.ch','posteo.co.uk','posteo.de','posteo.eu',"
-            "'posteo.lu','posteo.net','posteo.org','proton.me','protommail.com',"
-            "'protonmail.ch','pt.lu','riseup.net','runbox.com','secure.mailbox.org',"
-            "'startmail.com','system.li','temp.mailbox.org','t-online.de',"
-            "'tuta.io','tutamail.com','tutanota.com','tutanota.de','vodafone.de',"
-            "'vodafonemail.de','wanadoo.fr','web.de','xs4all.nl','yahoo.ca','yahoo.co.uk',"
-            "'yahoo.com','yahoo.de','yahoo.es','yahoo.fr','yahoo.it')"
+        email_domains_config_value = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("merge_duplicate_contacts.email_domains")
         )
 
+        if email_domains_config_value:
+            email_domains = email_domains_config_value.split(",")
```

Comment:
Well it's a manual configured setting. I wouldn't trust the correct spaceless format too much.

Better add the following line to delete whitespaces and ignore empty values.
```
email_domains = list(filter(None, (x.strip() for x in email_domains)))
```

Also this should be moved inside of the `if` if it's actually needed. No need to process it if the boolean isn't flagged.